torch/csrc/jit/codegen/cuda/kernel_ir.h

```diff
@@ -495,20 +400,13 @@ class TORCH_CUDA_CU_API Int final : public Val {
  public:
   using ScalarType = int64_t;
 
-  explicit Int(Passkey passkey, const c10::optional<ScalarType>& value)
-      : Val(passkey, ValType::Scalar, DataType::Int), maybe_value_(value) {}
+  explicit Int(
+      IrBuilderPasskey passkey,
+      const c10::optional<ScalarType>& value);
 
-  // SFINAE constructor to avoid 0 constant pointer ambiguity
-  template <
-      typename T,
-      typename = typename std::enable_if<
-          std::is_pointer<T>::value &&
-          std::is_convertible<T, const fuser::cuda::Int*>::value>::type>
-  explicit Int(Passkey passkey, T node)
-      : Val(passkey, ValType::Scalar, DataType::Int),
-        maybe_value_(node->value()) {
-    setName(node->name());
-  }
+  explicit Int(IrBuilderPasskey passkey, int value);
+
+  Int(IrBuilderPasskey passkey, const fuser::cuda::Int* stmt);
```

Comment:
Wouldn't there be a problem of overloading resolution as it was mentioned previously in the comment? Passing `0` would match with this `Int*` version as well as the above `int` version (and also `c10::optional` version?).

Reply:
I thought the precedence would route 0 directly to the `explicit Int(IrBuilderPasskey passkey, int value);` constructor as it wouldn't require any implicit casting.

Reply:
I believe this was an issue without such a constructor, which is why I added one. I had trouble getting the constructor out of this file with SFINAE which I was trying hard to do, but if you know how to do it please ;-)

Reply:
I think you're right. With the `int` version, it should be preferred over the other alternatives.

Reply:
Sorry, I also forgot that I got rid of this constructor later. It seemed very unnecessary as Int only has one member, so we can simply use the `c10::optional` constructor. We may even get rid of the `Statement*` constructors as those are only used during lowering (once Kernel uses IrContainer and can be copied directly from Fusion with IrCloner)